cmake/kokkos_functions.cmake

```diff
@@ -60,6 +60,278 @@ FUNCTION(kokkos_option CAMEL_SUFFIX DEFAULT TYPE DOCSTRING)
 
 ENDFUNCTION()
 
+FUNCTION(kokkos_append_config_line LINE)
+  SET(KOKKOS_TPL_EXPORTS "${KOKKOS_TPL_EXPORTS}\n${LINE}" PARENT_SCOPE)
```

Comment:
I'm a little worried about this `PARENT_SCOPE`, how do we handle multiple levels of directories?

Reply:
Yeah, we can change this to an internal cache variable.

Reply:
Done.